File: java/oop/InterfaceSample.java
package oop;

public class InterfaceSample {
    public static void main(String[] args) {
        System.out.println("light");
        CarLight carLight = new CarLight();
        carLight.turnOn();
        carLight.turnOff();
    }
    
}
interface Light {
    void turnOn();

    void turnOff();
}

class CarLight implements Light {

    @Override
    public void turnOn() {
        System.out.println("turn on");
    }

    @Override
    public void turnOff() {
        System.out.println("turn off");
    }
}

File: golang/oop/interface-sample.go
package main

import "fmt"

type light interface {
	TurnOn()
	TurnOff()
}

type CarLight struct{}

func (carLigth CarLight) TurnOn() {
	fmt.Println("trun on")
}
func (carLight CarLight) TurnOff() {
	fmt.Println("turn off")
}
func main() {
	var carLight CarLight
	fmt.Println("light")
	carLight.TurnOn()
	carLight.TurnOff()

}
